pkg/apis/seedmanagement/helper/helper.go

```diff
@@ -32,28 +34,37 @@ func GetBootstrap(bootstrap *seedmanagement.Bootstrap) seedmanagement.Bootstrap
 
 // IsMultiZonalManagedSeed checks if a managed seed is multi-zonal.
 func IsMultiZonalManagedSeed(managedSeed *seedmanagement.ManagedSeed) (bool, error) {
-	seedSpec, err := ExtractSeedSpec(managedSeed)
+	seedTemplate, _, err := ExtractSeedTemplate(managedSeed)
 	if err != nil {
 		return false, err
 	}
-	return seedSpec.HighAvailability != nil && seedSpec.HighAvailability.FailureTolerance.Type == gardencore.FailureToleranceTypeZone, nil
+	return seedTemplate.Spec.HighAvailability != nil && seedTemplate.Spec.HighAvailability.FailureTolerance.Type == gardencore.FailureToleranceTypeZone, nil
```

Comment:
Don't we need to check for the multi-zonal label here as well, i.e. the same logic as https://github.com/gardener/gardener/blob/39059990179c2c33cfa616d17d49c39701f521fb/pkg/apis/core/v1beta1/helper/helper.go#L1440?

Reply:
Yes that is correct. I really wish there was some place where common set of helper functions for all API could be kept. Will it be ok to create `pkg/apis/helper/helpers.go` and put all commonly used methods there?
To check a label all you need is either the v1.ObjectMeta or the label. Right now this needs to be repeated in multiple places which is not very nice.

Reply:
Hm, how would you envision to generically check `IsMultiZonalSeed`, independent from the API? Considering the labels is only one part, but the `.spec` is also curcial.